### DISCUSSION TITLE
Version 2.0.2 Released Today

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#1256] [General]
Blog post: https://www.oqtane.org/Resources/Blog/PostId/547/oqtane-202-released

Oqtane has been receiving plenty of exposure in recent months and was featured on the .NET Community Standup with Safia Abdalla and Jon Galloway on April 13. The 2.0.2 release contains a variety of user experience improvements for both software developers and end users.

Templates

One of the core philosophies of Oqtane is to be a "low ceremony" framework that simplifies the effort required in order for a developer to come up to speed with the framework. One of the best ways to achieve this goal is to provide functional templates that developers can utilize as the basis for their own solutions. A Module Creator was introduced in version 1.x to enable developers to quickly scaffold custom modules and the concept has been extended in the latest version with a Theme Creator as well as the ability to create custom local templates.

Themes and Containers

The ability to customize the look and feel of an application is an essential aspect to any framework, and Oqtane provides this capability through its theming feature. Inspired by a 2020 pull request from Pavel Veselý the framework has been enhanced with the ability for a theme to include custom settings for both theme and container components. The custom settings are surfaced as integrated tabs in the Oqtane user interface in the associated Page or Module settings areas. The Oqtane theme has been enhanced to take advantage of this feature and provide an example of how to implement it in your own themes.

SMTP Configuration

Sending notifications is a common feature in any application and Oqtane provides a robust capability for sending emails asynchronously using SMTP. This release introduces the capability to test your SMTP server configuration to ensure the settings are configured properly for your site.

Recycle Bin

Oqtane includes a recycle bin for restoring pages or modules which have previously been delete from your site. However in the event that you want to permanently delete them, Leigh Pointer submitted a pull request which allows the administrator to bulk remove all deleted pages and modules in a single operation.

This release includes 36 pull requests submitted by 6 different contributors, pushing the total number of project commits all-time to nearly 1800. The Oqtane framework continues to evolve at a rapid pace to meet the needs of .NET developers. The 2.0.2 release is available for download on Github.

**Comment by PavelVsl** [comment]
Thank you for implementing theme and container settings! It works!

**Comment by dtquocbao** [comment]
Just quick question: does this new release support External Template Module Creator only?

**Reply by leigh-pointer** [reply]
Yes, only external modules and themes.

**Reply by sbwalker** [reply]
@rockydant yes, the Internal Module Creator was removed as it is an anti-pattern. The whole point of a modular framework is to extend it with new functionality without modifying the core framework. The Internal Module Creator was adding code to the core framework which breaks this rule.
